src/commands/plug.rs

```diff
@@ -0,0 +1,132 @@
+use std::{io::Write as _, path::PathBuf};
+
+use anyhow::{bail, Context as _, Result};
+use clap::Args;
+use wac_graph::{CompositionGraph, EncodeOptions, NodeId, PackageId};
+use wac_types::{Package, SubtypeChecker};
+
+/// Plugs the exports of any number of 'plug' components into the imports of a 'socket' component.
+#[derive(Args)]
+#[clap(disable_version_flag = true)]
+pub struct PlugCommand {
+    /// The path to the plug component.
+    ///
+    /// More than one plug can be supplied.
+    #[clap(long = "plug", value_name = "PLUG_PATH", required = true)]
+    pub plugs: Vec<PathBuf>,
+
+    /// The path to the socket component
+    #[clap(value_name = "SOCKET_PATH", required = true)]
+    pub socket: PathBuf,
+
+    /// The path to write the output to.
+    ///
+    /// If not specified, the output will be written to stdout.
+    #[clap(long, short = 'o')]
+    pub output: Option<PathBuf>,
+}
+
+impl PlugCommand {
+    /// Executes the command.
+    pub async fn exec(&self) -> Result<()> {
+        log::debug!("executing plug command");
+        let mut graph = CompositionGraph::new();
+
+        let socket = std::fs::read(&self.socket).with_context(|| {
+            format!(
+                "failed to read socket component `{path}`",
+                path = self.socket.display()
+            )
+        })?;
+
+        let socket = Package::from_bytes("socket", None, socket, graph.types_mut())?;
+        let socket = graph.register_package(socket)?;
+        let socket_instantiation = graph.instantiate(socket);
+
+        // Plug each plug into the socket.
+        for (i, plug) in self.plugs.iter().enumerate() {
+            let plug = std::fs::read(&plug).with_context(|| {
+                format!(
+                    "failed to read plug component `{path}`",
+                    path = plug.display()
+                )
+            })?;
+            plug_into_socket(
+                &format!("plug{i}"),
```

Comment:
Perhaps we should use a (sanitized) file stem as the name of package? Eventually we will be validating the package names in the constructor (see #90). Something like `plug:<filestem>` and add a numerical suffix if that's not unique?

Reply:
Granted, the package names aren't terribly relevant currently, but I could see a `--import-plugs` option to this command in the future that encodes with component imports rather than defining them.

Reply:
Done - though I didn't do any package name sanitizing (I couldn't find anywhere where we're currently doing that) so I just added a todo comment. Is that fine?

Reply:
Totally fine, we'll add that when we fix #90.